inception-ui-external-search/src/main/java/de/tudarmstadt/ukp/inception/app/ui/externalsearch/DocumentImporter.java

```diff
@@ -0,0 +1,71 @@
+package de.tudarmstadt.ukp.inception.app.ui.externalsearch;
+
+import de.tudarmstadt.ukp.clarin.webanno.api.DocumentService;
+import de.tudarmstadt.ukp.clarin.webanno.model.Project;
+import de.tudarmstadt.ukp.clarin.webanno.model.SourceDocument;
+import de.tudarmstadt.ukp.clarin.webanno.security.UserDao;
+import de.tudarmstadt.ukp.clarin.webanno.security.model.User;
+import de.tudarmstadt.ukp.inception.externalsearch.ExternalSearchResult;
+import de.tudarmstadt.ukp.inception.externalsearch.ExternalSearchService;
+import de.tudarmstadt.ukp.inception.externalsearch.model.DocumentRepository;
+import org.apache.commons.lang3.exception.ExceptionUtils;
+import org.apache.uima.UIMAException;
+import org.apache.wicket.spring.injection.annot.SpringBean;
+
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.Serializable;
+import java.nio.charset.StandardCharsets;
+
+public class DocumentImporter implements Serializable
+{
+
+    private static final String PLAIN_TEXT = "text";
+
+    private ExternalSearchService externalSearchService;
+
+    private DocumentService documentService;
+
+    private User user;
+
+    private Project project;
+
+
+    public DocumentImporter(ExternalSearchService aExternalSearchService,
+        DocumentService aDocumentService, User aUser,Project aProject)
+    {
+        externalSearchService = aExternalSearchService;
+        documentService = aDocumentService;
+        user = aUser;
+
+        project = aProject;
+    }
+
+    public void importDocumentFromDocumentRepository(String aDocumentTitle,
+        DocumentRepository aRepository) throws IOException
+    {
+        String text = externalSearchService.getDocumentById(user, aRepository, aDocumentTitle)
+            .getText();
+
+        if (documentService.existsSourceDocument(project, aDocumentTitle)) {
```

Comment:
I'd respectfully disagree. A boolean is unable to carry information on what the problem was and why the import failed. If you ask me, better thrown an exception and let the calling code decide what to do with it (display a helpful error message to the user including a reason for the failure, handle it in some other way).

Reply:
Ah - I would be inclined to make the distinction between "We attempted the import and it failed" (which should definitely be an exception) and "we didn't do anything because the document had already been imported", in which case an exception seems weird, since the expected post-condition that "the document is in the database" still holds true.  But I'm happy to defer to the project maintainer. :-)

Reply:
Modified so that if document already imported, it will return false instead of throwing an exception. The other exceptions remain unchanged. If successful, returns true.